controllers/depresolver/depresolver.go

```diff
@@ -139,6 +139,9 @@ type Config struct {
 	MetricsAddress string `env:"METRICS_ADDRESS, default=0.0.0.0:8080"`
 	// extDNSEnabled hidden. EdgeDNSType defines all enabled Enabled types
 	extDNSEnabled bool `env:"EXTDNS_ENABLED, default=false"`
+	// Route53HostedZoneID identifier of route53 hosted zone that's added (if not empty)
+	// for external-dns deployment as part of the txt-owner-id
+	Route53HostedZoneID string `env:"ROUTE53_HOSTED_ZONE_ID"`
```

Comment:
What about NS1 or rfc2136 provider, IIUC controller now expects R53 hosted zone ID and will fail if string matching pattern won't be provided.

Reply:
It can be empty, the regexp check is done only if it's not an empty string. So for NS1 or rfc2136 the env var is not going to be set.

If it's empty the DNS_ZONE value is used instead (same as before in the _helpers.tpl)

Reply:
Maybe we can use same owner across providers? Not sure mainaining separate case for r53 worth it.